readiness-tool/popup.js

```diff
@@ -0,0 +1,233 @@
+
+self.popups = {};
+self.popups.isSupported = isSupported;
+self.popups.doesRegexExist = doesRegexExist;
+
+/**
+   * Callback function that sends a response upon receiving message
+   * @param {!Object} request - Message Object
+   * @param {!Object} sender - Message sender defined here
+   */
+chrome.runtime.onMessage.addListener(function (request, sender) {
+    if (request.action == "getSource") {
+        // htmlOfTab contains the page source as a string
+        htmlOfTab = request.source;
+        findDetectedApps(htmlOfTab);
+    }
+});
+
+/** Event called when window is loaded */
+function onWindowLoad() {
+    var htmlOfTab;
+    // When page is loaded, display "Loading..." so the user expects content
+    var supportedAds = document.getElementById("ads-supported");
+    var supportedAnalytics = document.getElementById("analytics-supported");
+    var notSupportedAds = document.getElementById("ads-notSupported");
+    var notSupportedAnalytics = document.getElementById("analytics-notSupported");
+    var notSupportedOther = document.getElementById("other-notSupported");
+    supportedAds.innerHTML = 'Loading...';
+    supportedAnalytics.innerHTML = "Loading...";
+    notSupportedAds.innerHTML = "Loading...";
+    notSupportedAnalytics.innerHTML = "Loading...";
+    notSupportedOther.innerHTML = "Loading...";
+    
+    // Gets the DOM of the current web page and converts it to a string
+    chrome.tabs.executeScript(null, {
+        file: "getPagesSource.js"
+    }, function () {});
+}
+
+/**
+   * Returns all the 3rd party applications found on the website
+   * @param {String} html - String containing all HTML on the page
+   * @return {Object} 
+   */
+function findDetectedApps(html) {
+    var htmlString = html;
+    let linkToApps = (chrome.runtime.getURL('apps.json'));
```

Comment:
Oh, so ES6 syntax like `let` works? 👍 

Please use `const` for variables whose value doesn't change and `let` otherwise. `var` is discouraged: https://google.github.io/styleguide/jsguide.html#features-use-const-and-let